### ISSUE TITLE
brew bump --cask no longer works on linux

### ISSUE THREAD

**Issue by hybras** [Homebrew/brew#14341]
### `brew doctor` output

```shell
Your system is ready to brew.
```

### Verification

- [X] My "`brew doctor` output" above says `Your system is ready to brew.` and am still able to reproduce my issue.
- [X] I ran `brew update` twice and am still able to reproduce my issue.
- [X] This issue's title and/or description do not reference a single formula e.g. `brew install wget`. If they do, open an issue at https://github.com/Homebrew/homebrew-core/issues/new/choose instead.

### `brew config` output

```shell
HOMEBREW_VERSION: 3.6.5
ORIGIN: https://github.com/Homebrew/brew
HEAD: fb52a12144e80118c112ea744ab65ab6ca997d67
Last commit: 3 months ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: b227058ed19b12a35f909a7cabe807592bc0d927
Core tap last commit: 3 months ago
Core tap branch: master
HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 4
HOMEBREW_NO_AUTO_UPDATE: set
Homebrew Ruby: 2.6.8 => /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8_1/bin/ruby
CPU: quad-core 64-bit unknown_0x0_0x0
Clang: N/A
Git: 2.38.0 => /bin/git
Curl: 7.81.0 => /bin/curl
Kernel: Linux 5.15.49-linuxkit x86_64 GNU/Linux
OS: Unknown
Host glibc: 2.35
/usr/bin/gcc: 11.2.0
/usr/bin/ruby: N/A
glibc: N/A
gcc@11: N/A
gcc: N/A
xorg: N/A
```

### What were you trying to do (and why)?

I'm trying to run `brew bump` on casks and formula in CI. I'd like my package versions to update as automatically as possible. My goal was to replicate homebrew's system of package updates for my tap.

This issue appears to be a consequence of #13941. 

### What happened (include all command output)?

My casks are no longer detected.

```
Error: No available formula with the name "kdeconnect". Did you mean kt-connect?
```

### What did you expect to happen?

A new version would be detected using livecheck, the package manifest would be edited to include it, and  pr would be filed with my repo.

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
brew tap hybras/tap # or any tap with casks
brew bump --open-pr kdeconnect # or on any cask
```

**Comment by apainintheneck** [comment]
This is expected behavior as result of merging in https://github.com/Homebrew/brew/pull/14206 but unfortunately this was one of the edge cases that could crop up.

That being said the `brew bump` command calls either `brew bump-cask-pr` or `brew bump-formula-pr` internally and you should still be able to to use those commands directly after that update. Would that work for your use case instead?

**Comment by hybras** [comment]
It does not. bump-cask-pr called audit which has the --formula flag passed to it as part of the aforementioned change

**Comment by apainintheneck** [comment]
I forgot about `brew bump-cask-pr` shelled out to `brew audit` internally. That's unfortunate. I'll take a look and see how difficult it would be to make an exception for a few dev commands.

**Comment by apainintheneck** [comment]
@hybras I just merged in #14344 which should solve this issue. Let me know if anything else comes up.

**Comment by hybras** [comment]
Thank you for your prompt response and understanding of my situation
